Wangscape/noise/module/Bitmap.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+#include <memory>
+#include <string>
+#include <noise/noise.h>
+#include <SFML/Graphics.hpp>
+
+namespace noise
+{
+namespace module
+{
+
+// Noise module that takes values from a bitmap.
+// The bitmap file should be a greyscale image.
+// Only red values will be used.
+class Bitmap : public Module
+{
+public:
+    Bitmap();
+    virtual int GetSourceModuleCount() const
+    {
+        return 0;
+    }
+    const std::string& GetFilename() const;
+    void SetFilename(const std::string& filename);
+
+    const sf::Vector2u& GetResolution() const;
+    void SetResolution(const sf::Vector2u& resolution);
+
+    sf::Rect<double> GetRegion() const;
+    void SetRegion(sf::Rect<double> region);
+
+    double GetDefaultValue() const;
+    void SetDefaultValue(double defaultValue);
+
+
+    virtual double GetValue(double x, double y, double z) const;
+private:
+    double getPixel(size_t x, size_t y) const;
+    std::string  mFilename;
```

Comment:
This `mFilename` bothers me. Why is it needed? What's a purpose of saving it? By looking at an interface of this class it's hard for me to guess that this class beside storing an image also loads it and it does so in the `SetFilename()` function - it seems to be only a setter of `mFilename` variable and I'd suppose that there would be also a `SetImage` function.

You're caching these images in `loadedImages`, so maybe there's no need to store `mImage` in an instance of this class (which of course would be weird in **Bitmap** :wink:), only a filename - key of that map.